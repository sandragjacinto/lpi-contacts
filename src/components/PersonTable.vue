<template>
    <table  class="table is-bordered is-striped is-narrow is-hoverable is-fullwidth">
    <thead>
        <tr>
        <th>Name</th>
        <th>Age</th>
        <th>Email</th>
        <th>City</th>
        <th>Country</th>
        </tr>
    </thead>
    <tbody>
        <tr v-for="person in peopleList" :key="person.id" @click="showModal(person)">
        <td>{{ person.firstname }}</td>
        <td>{{ person.lastname }}</td>
        <td>{{ person.email }}</td>
        <td>{{ person.address.city }}</td>
        <td>{{ person.address.country }}</td>
        </tr>
    </tbody>
    </table>
</template>
  
<script >

export default {
  props: {
    personList: {
      type: Array,
     default: []
    }
  },
};
</script>