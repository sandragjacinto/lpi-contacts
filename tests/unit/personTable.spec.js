import { shallowMount } from '@vue/test-utils'
import PersonTable from '@/components/PersonTable.vue'

const peopleList = [{
  firstname: 'test',
  lastname: 'testLast',
  email: 'test@test.com',
  address: {
    city: 'paris',
    country: 'france'
  }
}]

describe('PersonTable.vue', () => {
  it('sould render the list of persons', () => {
    const { element } = shallowMount(PersonTable, {
      props: { peopleList }
    })
    expect(element).toMatchSnapshot()
  })
})
