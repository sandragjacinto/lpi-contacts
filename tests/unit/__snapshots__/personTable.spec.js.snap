// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonTable.vue sould render the list of persons 1`] = `
<table
  class="table is-bordered is-striped is-narrow is-hoverable is-fullwidth"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Age
      </th>
      <th>
        Email
      </th>
      <th>
        City
      </th>
      <th>
        Country
      </th>
    </tr>
  </thead>
  <tbody>
    
    <tr>
      <td>
        test
      </td>
      <td>
        testLast
      </td>
      <td>
        test@test.com
      </td>
      <td>
        paris
      </td>
      <td>
        france
      </td>
    </tr>
    
  </tbody>
</table>
`;
